File: cross_lang/algorithms/sieve.js
function count_primes(limit) {
    const flags = new Array(limit + 1).fill(1);
    for (let p = 2; p * p <= limit; p++) {
        if (flags[p] === 1) {
            for (let multiple = p * p; multiple <= limit; multiple += p) {
                flags[multiple] = 0;
            }
        }
    }

    let count = 0;
    for (let j = 2; j <= limit; j++) {
        if (flags[j] === 1) {
            count++;
        }
    }
    return count;
}

const primes = count_primes(50000);
console.log(primes);

File: cross_lang/algorithms/sieve.py
def count_primes(limit):
    flags = [1] * (limit + 1)
    p = 2
    while p * p <= limit:
        if flags[p] == 1:
            multiple = p * p
            while multiple <= limit:
                flags[multiple] = 0
                multiple += p
        p += 1

    count = 0
    for j in range(2, limit + 1):
        if flags[j] == 1:
            count += 1
    return count

if __name__ == "__main__":
    primes = count_primes(50000)
    print(primes)
